### PR TITLE
Refactored Server Instance Tests

### PR DESCRIPTION
**Pull request by VasuBhog**
This PR refactors the Server Instance tests by removing the helper methods and changing the test to use Sinon.

### CODE REVIEW

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 1]
```diff
@@ -153,32 +97,52 @@ export class PerFolderServerInstance implements IServerInstance {
 		await this.startInternal();
 	}
 
+	public spawn(command: string, args?: ReadonlyArray<string>, options?: SpawnOptions): ChildProcess {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts:100]: Why not just remove this completely and call spawn directly from whatever is calling this?

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 2]
```diff
@@ -153,32 +97,52 @@ export class PerFolderServerInstance implements IServerInstance {
...
+		return spawn(command, args, options);
+	}
+
+	public ensureProcessEnded(childProcess: ChildProcess): void {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts:104]: Why is this part of the class? It doesn't need access to any class members so should just be a helper function somewhere I'd say.

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 3]
```diff
@@ -25,6 +25,26 @@ const defaultPort = 8888;
 type MessageListener = (data: string | Buffer) => void;
 type ErrorListener = (err: any) => void;
 
+/**
+ * Helper function to enable testing
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts:29]: This function isn't just used for testing. Please update to be descriptive of what the function is actually doing.

File: extensions/notebook/src/jupyter/serverInstance.ts [hunk 4]
```diff
@@ -122,14 +88,13 @@ export class PerFolderServerInstance implements IServerInstance {
 	private _uri: vscode.Uri;
 	private _isStarted: boolean = false;
 	private _isStopping: boolean = false;
-	private utils: ServerInstanceUtils;
 	private childProcess: ChildProcess;
 	private errorHandler: ErrorHandler = new ErrorHandler();
+	// private utils: ServerInstanceUtils;
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/serverInstance.ts:93]: remove

File: extensions/notebook/src/jupyter/serverInstance.ts (hunk 3 above)
Comment (VasuBhog) [reply extensions/notebook/src/jupyter/serverInstance.ts:29]: Gotcha. Would this be descriptive enough?
* Helper function ensures server instance process stops and enable testing

**Pull request merged**